### PR TITLE
Embed base execution policies to datapar execution policies

### PR DESCRIPTION
**Pull request by srinivasyadav18**
## Proposed Changes

  - Removed datapar/execution_policy.hpp inclusion
  - Embedded  base execution policies for datapar execution policies

### CODE REVIEW

File: libs/core/executors/include/hpx/executors/datapar/execution_policy.hpp [hunk 1]
```diff
@@ -137,6 +138,9 @@ namespace hpx { namespace execution { inline namespace v1 {
                     HPX_FORWARD(Parameters, params)...));
         }
 
+        using base_policy_type = sequenced_task_policy;
+        static constexpr base_policy_type base_policy = sequenced_task_policy{};
```
Comment (hkaiser) [inline libs/core/executors/include/hpx/executors/datapar/execution_policy.hpp:142]: I'd suggest not to include a pre-constructed instance of the base policy. For one, it's trivial to construct one if needed, but more importantly (and mostly for the _shim policies), constructing a new instance requires access to the concrete executor and parameters objects of the source policy. Those however are not necessarily known at compile time.

File: libs/core/executors/include/hpx/executors/execution_policy.hpp [hunk 2]
```diff
@@ -1753,4 +1752,4 @@ namespace hpx { namespace detail {
     };
 
     /// \endcond
-}}    // namespace hpx::detail
+}}    // namespace hpx::detail
```
Comment (hkaiser) [inline libs/core/executors/include/hpx/executors/execution_policy.hpp:1755]: Please add a newline character at the end of the file.

File: libs/core/executors/include/hpx/executors/execution_policy.hpp [hunk 3]
```diff
@@ -18,7 +18,6 @@
 #include <hpx/execution/traits/is_execution_policy.hpp>
 #include <hpx/execution_base/traits/is_executor.hpp>
 #include <hpx/execution_base/traits/is_executor_parameters.hpp>
-#include <hpx/executors/datapar/execution_policy.hpp>
```
Comment (hkaiser) [inline libs/core/executors/include/hpx/executors/execution_policy.hpp:21]: Excellent catch!

File: libs/core/executors/include/hpx/executors/datapar/execution_policy.hpp [hunk 4]
```diff
@@ -290,8 +293,11 @@ namespace hpx { namespace execution { inline namespace v1 {
 
         using base_policy_type =
             sequenced_task_policy_shim<Executor, Parameters>;
-        static constexpr base_policy_type base_policy =
-            sequenced_task_policy_shim<Executor, Parameters>{};
+        base_policy_type base_policy()
+        {
+            return simd_task_policy_shim<Executor, Parameters>(
+                HPX_FORWARD(Executor, exec_), HPX_FORWARD(Parameters, params_));
+        }
```
Comment (hkaiser) [inline libs/core/executors/include/hpx/executors/datapar/execution_policy.hpp:300]:
```suggestion
        base_policy_type base_policy()
        {
            return base_policy_type(exec_, params_);
        }
```
We should not forward the members to the constructor of the new instance, those are not forwarding references,

Comment (srinivasyadav18) [reply libs/core/executors/include/hpx/executors/datapar/execution_policy.hpp:300]: Oops!, missed it completely. Will resolve it ASAP!,
Thanks! :)

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!

**Comment by hkaiser** [comment]
bors merge

**Pull request merged**